src/playground/sample_data_ingestion/utils/sample_data_ingestor/tests/sample_data_ingestor_unit_test.py

```diff
@@ -0,0 +1,119 @@
+# opensearch-py==2.0.0
+from opensearchpy import OpenSearch
+
+# Standard libraries
+from json import dumps, loads
+import pytest
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from sample_data_ingestor import ingest
+
+# Constants
+INDEX_NAME = "ingest-test"
+DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+INDEX_BODY = {
+  'settings': {
+    'index': {
+      'number_of_shards': 1
+    }
+  }
+}
+
+
+# Configurations for connection (NEED TO CHANGE)
+HOST = os.environ.get("HOST")
+PORT = 9200
+AUTH = (os.environ.get("SAMPLE_DATA_USERNAME"), os.environ.get("SAMPLE_DATA_PASSWORD"))
+
+# Establish connection with OS
+client = OpenSearch(
+    hosts = [{'host': HOST, 'port': PORT}],
+    http_compress = True,
+    http_auth = AUTH,
+    use_ssl = True,
+    verify_certs = False,
+    ssl_assert_hostname = False,
+    ssl_show_warn = False
+)
+
+
+# Sample inputs (valid)
+jinput = {
+        "year": "year",
+        "random number": "integer",
+        "email": "email",
+        }
+
+jinput2 = {"properties": {
+        "first name":    { "type" : ["first_name"]},
+        "last name":     { "type" : "last_name"},
+        "companies":{ "type" : ["array", "company", 5] }
+        }}
+
+jinput3 = dumps({
+        "year": "year",
+        "addresses": ["array", "address", "integer", [1, 3]],
+        "director": "name"
+        })
+
+jinput4 = dumps({"properties": {
+        "year":    { "type" : ["year"]},
+        "addresses":     { "type" : ["array", "address", "integer", (1, 3)] },
+        "director":{ "type" : "name" }
+        }})
+
+jinput5 = dumps({"properties": {
+        "year":    { "type" : ["year"]},
+        "integers":     { "type" : ["array", "integer", "integer", (1, 3), 5, 10] },
+        "director":{ "type" : "name" }
+        }})
+
+jinput6 = {
+        "date": "unix_time",
+        "average cpu usage": ["integer", 20, 30]
+}
+
```

Comment:
Can we move the test cases to seperate file to make it modularized?

Reply:
Done, the inputs are located in the file `valid-template-inputs.json`